base/asyncmap.jl

```diff
@@ -22,17 +23,17 @@ end
 
 function AsyncCollector(f, results, c...; ntasks=0)
     if ntasks == 0
-        ntasks = 100
+        ntasks = max(nworkers(), 100)
```

Comment:
@bjarthur this should be the cause of your observation of parallelism being limited to 100 workers.